compiler/modules/nback/number.bal

```diff
@@ -199,36 +218,27 @@ function buildConvertToFloat(llvm:Builder builder, Scaffold scaffold, bir:Conver
         buildConvertIntToFloat(builder, scaffold, val, insn);
         return;
     }
-    else if repr.base != BASE_REPR_TAGGED {
-        return scaffold.unimplementedErr("convert form decimal to float", insn.pos);
+    else if repr.base == BASE_REPR_TAGGED && repr.subtype == t:DECIMAL {
+        buildConvertDecimalToFloat(builder, scaffold, val, insn);
+        return;
     }
     // convert to float form tagged pointer
 
     // number part of semType must be some *non-empty* combination of
     // (some or all of) int, float and decimal
```

Comment:
Let's remove this comment, it's no longer useful.

Reply:
Removed